### PR TITLE
Fixes #42; slightly nicer error message when no 'code' field.

### LINKED ISSUE #42: when 'deny' is clicked on google plugin, the result is a pretty undescriptive error about a missing field
to repro:

1. attempt to authenticate with auth2google plugin
2. select an account
3. click 'deny'
4. observe error.

### PR DESCRIPTION
**Pull request by silky**
- permission denied when no "required" fields (code/state)
- fix old url in repo source
- deleted unused import

### CODE REVIEW

File: Yesod/Auth/OAuth2.hs [hunk 1]
```diff
@@ -102,15 +102,18 @@ authOAuth2Widget widget name oauth getCreds = AuthPlugin name dispatch login
...
-                code <- lift $ runInputGet $ ireq textField "code"
-                oauth' <- withCallback csrfToken
-                master <- lift getYesod
-                result <- liftIO $ fetchAccessToken (authHttpManager master) oauth' (encodeUtf8 code)
-                case result of
-                    Left _ -> permissionDenied "Unable to retreive OAuth2 token"
-                    Right token -> do
-                        creds <- liftIO $ getCreds (authHttpManager master) token
-                        lift $ setCredsRedirect creds
+                mcode <- lift $ runInputGet $ iopt textField "code"
```
Comment (pbrisbin) [inline Yesod/Auth/OAuth2.hs]: This code's getting a little noisy with all the nested `case`s.

What do you think about wrapping `lookupGetParam` into a `requireGetParam` to cetralize the `Maybe` handling out of the main logic, then relying on `permissionDenied`'s short-circuit nature to bail early:

```hs
dispatch "GET" ["callback"] = do
    code <- requireGetParam "code"
    csrfToken <- requireGetParam "state"

    unless (csrfToken == oldToken) $ permissionDenied "Invalid OAuth2 state token"

    -- main logic...

  where
    requireGetParam key =
        maybe (permissionDenied $ key <> " parameter not provided") return <$> lookupGetParam key
```

Note: not test-compiled

File: yesod-auth-oauth2.cabal [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name:            yesod-auth-oauth2
-version:         0.1.4
+version:         0.1.5
```
Comment (pbrisbin) [inline yesod-auth-oauth2.cabal]: Can you back out the version bump? I can take care of that separately after we merge.

File: yesod-auth-oauth2.cabal [hunk 3]
```diff
@@ -52,4 +52,4 @@ library
 
 source-repository head
   type:     git
-  location: https://github.com/thoughtbot/authenticate-oauth2.git
+  location: https://github.com/thoughtbot/yesod-auth-oauth2.git
```
Comment (pbrisbin) [inline yesod-auth-oauth2.cabal]: Nice catch! Do you might making this tweak its own commit?

**Comment by silky** [comment]
thanks for the feedback; all good comments, i will fix all these things and resubmit.

**Comment by pbrisbin** [comment]
Sorry for dropping the ball on this. I'll spend some time tomorrow going back over this and hopefully merging / releasing. Thanks for the contribution.

**Comment by silky** [comment]
no stress mate :) no probs!

**Comment by pbrisbin** [comment]
@jferris this LGTM. If you have no objections, I'll clean up the history a bit and merge / release (maybe with an hlint pass, I spy some redundant parens).

**Comment by jferris** [comment]
This looks good to me.

**Comment by pbrisbin** [comment]
Merged as 7354c36e1326d298e543fa65cf226153ed4a8a0b
